src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -268,7 +268,11 @@ public function replaceArgument($index, $argument)
             throw new OutOfBoundsException(sprintf('Cannot replace arguments for class "%s" if none have been configured yet.', $this->class));
         }
 
-        if (\is_int($index) && ($index < 0 || $index > \count($this->arguments) - 1)) {
+        /** @var int[] $intIndices */
+        $intIndices = array_filter(array_keys($this->arguments), function ($index) {
+            return \is_int($index);
+        });
+        if (\is_int($index) && ($index < 0 || $intIndices && $index > max($intIndices))) {
```

Comment:
I'm wondering, is this check even necessary considering the `array_key_exists()` call right after?